src/components/Explore/GroupBySelector.tsx

```diff
@@ -45,37 +80,64 @@ export function GroupBySelector({ options, radioAttributes, value, onChange, sho
+  const filterValidOptions = useMemo(() => {
+    const isOptionValid = (option: string) => {
+      // Check if option exists in dropdown
+      let isValid = !!options.find((o) => o.value === option);
 
-        // remove radio options that are in the filters
-        if (filters.find((f) => f.key === op && (f.operator === '=' || f.operator === '!='))) {
-          return false;
-        }
+      // Remove options that are in the filters
+      if (filters.find((f) => f.key === option && (f.operator === '=' || f.operator === '!='))) {
+        return false;
+      }
 
-        // if filters (primary signal) has 'Full Traces' selected, then don't add rootName or rootServiceName to options
-        // as you would overwrite it in the query if it's selected
-        if (filters.find((f) => f.key === 'nestedSetParent')) {
-          checks = checks && op !== 'rootName' && op !== 'rootServiceName';
-        }
+      // If filters (primary signal) has 'Full Traces' selected, then don't add rootName or rootServiceName to options
+      if (filters.find((f) => f.key === 'nestedSetParent')) {
+        isValid = isValid && option !== 'rootName' && option !== 'rootServiceName';
+      }
 
-        // if rate or error rate metric is selected, then don't add status to options
-        // as you would overwrite it in the query if it's selected
-        if (metricValue === 'rate' || metricValue === 'errors') {
-          checks = checks && op !== 'status';
-        }
+      // If rate or error rate metric is selected, then don't add status to options
+      if (metricValue === 'rate' || metricValue === 'errors') {
+        isValid = isValid && option !== 'status';
+      }
+
+      // Filter out ignored attributes
+      isValid = isValid && !ignoredAttributes.includes(option);
+
+      return isValid;
+    };
 
-        return checks;
-      })
+    const allOptionValues = options.map(op => op.value?.toString() ?? '');
+    
+    const validRadioOptions = radioAttributes.filter(isOptionValid);
+    
+    // Get valid other options (not in radio attributes, but valid and have been clicked)
+    const validOtherOptions = allOptionValues
+      .filter(option => 
+        isOptionValid(option) && 
+        !radioAttributes.includes(option) && 
+        (clickCounts[option] || 0) > 0
+      );
+
+    const allOptions = [...validRadioOptions, ...validOtherOptions];
+
+    return {
+      allOptions,
+      isOptionValid
+    };
+  }, [options, radioAttributes, filters, metricValue, clickCounts]);
+
+  const radioOptions = useMemo(() => {
+    let radioOptionsWidth = 0;
+    const { allOptions } = filterValidOptions;
+    
+    return allOptions
       .map((attribute) => ({
         label: attribute.replace(SPAN_ATTR, '').replace(RESOURCE_ATTR, ''),
         text: attribute,
         value: attribute,
+        clickCount: clickCounts[attribute] || 0
       }))
+      .sort((a, b) => b.clickCount - a.clickCount)
```

Comment:
It works as described, though personally I find it a bit unclear what the behavior is (not knowing it upfront). The items when clicked start to jump around and it might be hard to figure out what is going on based on UI. A few things that would help me:

- A tooltip describing how options are sorted
- Sorting only on load (e.g. after reload) not on each click to reduce items jumping back and forth

Alternatively we could allow manually "pinning" items, or dismissing items instead of promoting them but that would require a lot of UI changes.

Reply:
This is a good tip, will look into it.